Update binplacing paths and default BuildConfiguration.

mellinoe opened the pull request:
* These now match the spec, at least for Windows.
* Native shims are now binplaced to a location that corresponds to the spec'd location. We will need to make sure this location stays consistent between the two build systems.
* We are still using "netcoreapp1.1" as the default TargetGroup. Eventually this should just be "netcoreapp", but we need to modify all of the projct files to make that change.

@weshaggard @joperezr @ericstj

joperezr commented on binplace.targets:
```diff
@@ -31,32 +31,24 @@
-    <ItemGroup Condition="'$(NuGetTargetMoniker)' == '' Or $(NuGetTargetMoniker.Contains('NETCoreApp')) Or $(NuGetTargetMoniker.Contains('NETStandard'))">
-      <TargetingPackDir Include="$(BinDir)netcoreapp\TargetingPack" />
+    <ItemGroup>
+      <TargetingPackDir Include="$(BinDir)ref\$(BuildConfiguration)" />
```
Why do we still need to calculate this properties inside a Target? can't we just pomote them to a static property?

mellinoe replied:
Yeah we should be able to do that, I think.

weshaggard replied:
I updated this with my PR. https://github.com/dotnet/corefx/pull/14550

mellinoe commented on binplace.targets:
```diff
@@ -31,32 +31,24 @@
-    <ItemGroup Condition="'$(NuGetTargetMoniker)' == '' Or $(NuGetTargetMoniker.Contains('NETCoreApp')) Or $(NuGetTargetMoniker.Contains('NETStandard'))">
-      <RuntimeDir Include="$(BinDir)netcoreapp\Runtime" />
+    <ItemGroup>
+      <RuntimeDir Include="$(BinDir)runtime\$(BuildConfiguration)" />
```
This is actually a different ItemGroup with the same name, because I initially thought we were going to need to resolve from multiple directories (with the actual runtime and Xunit in separate directories). I should probably rename this property in some way to make that clear.

mellinoe replied:
(Or just use `$(RuntimeDir)` directly I guess).

weshaggard replied:
Don't worry about this we have plenty of clean-up left here and I've done part of it with https://github.com/dotnet/corefx/pull/14550.

weshaggard commented on dir.props:
```diff
@@ -22,7 +22,7 @@
     <SkipImportCodeAnalysisTargets>true</SkipImportCodeAnalysisTargets>
     <!-- Hacky BuildConfiguration -->
-    <BuildConfiguration Condition="'$(BuildConfiguration)' == ''">netcoreapp1.1-$(OSEnvironment)</BuildConfiguration>
+    <BuildConfiguration Condition="'$(BuildConfiguration)' == ''">netcoreapp1.1-$(OSEnvironment)-x64-Debug</BuildConfiguration>
```
This should be "-Debug-x64" for consistency with the spec. I suspect what you have will actually still work but I would like to keep them consistent.

weshaggard commented on src/Native/build-native.cmd:
```diff
@@ -9,6 +9,8 @@ set __rootDir=%~dp0..\..
 set __CMakeBinDir=""
 set __IntermediatesDir=""
 set __BuildArch=x64
+:: TODO: This should default to 'netcoreapp' whenever the managed assemblies do the same.
```
You can actually change the managed side to netcoreapp today. I already did with my recent PR https://github.com/dotnet/corefx/pull/14550.

weshaggard commented:
The changes look reasonable a number of them conflict with my PR https://github.com/dotnet/corefx/pull/14550. I think it is best to just update the native side with your change, but do go ahead and just use netcoreapp for the TargetGroup. Things will align once my change goes in as well.

mellinoe commented:
@weshaggard I removed the changes except for the changes to build-native.cmd/sh. And I fixed the paths so that the order is correct (`-Debug-x64` on the end).
